test/user/run.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/sh
+
+set -e
+
+cd $(dirname $0)
+. ../util.sh
+
+docker compose up -d
+sleep 15 
+
+docker compose exec backup backup
+sudo cp -r $(docker volume inspect --format='{{ .Mountpoint }}' commands_archive) ./local
+
+tar -xvf ./local/test.tar.gz
+if [ ! -f ./backup/data/dump.zip ]; then
+  fail "Could not find file written by pre command."
+fi
+pass "Found expected file."
+
+if [ -f ./backup/data/not-relevant.txt ]; then
+  fail "Command ran for container with other label."
+fi
+pass "Command did not run for container with other label."
+
+if [ -f ./backup/data/post.txt ]; then
+  fail "File created in post command was present in backup."
+fi
+pass "Did not find unexpected file."
```

Comment:
How does this test the code? Is there any way for those files end up in ./backup/data, unless you don't put them in the folder on purpose?

Reply:
The files are written by the commands under test and then archived. If the commands don't run and the files don't get created, the `-f` checks for existence above will fail.